### ISSUE TITLE
TypeScript: Odd inline intersection types formatting

### ISSUE THREAD

**Issue by prencher** [prettier/prettier#1880]
Input:
```ts
declare const window: Window & {
  __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: Function;
};
```
Output with `--trailing-comma all`:
```ts
declare const window: Window &
  {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__: Function,
  };
```
Would expect this to retain the formatting. Also note that this is an interface and it is incorrectly replacing the semicolon with a comma, similar to #1879. Without `--trailing-comma` it correctly strips the semicolon.

**Comment by vjeux** [comment]
We retain this formatting for flow types, looks like my attempt at making it work for TS is not working. I'll look into it.

**Comment by vjeux** [comment]
Seriously, awesome job with those bug reports! Thanks!

**Comment by vjeux** [comment]
I just published 1.4.1 which fixes it!
